backend/de.metas.dunning/src/main/java/de/metas/dunning/api/impl/DefaultDunningProducer.java

```diff
@@ -164,6 +175,24 @@ protected I_C_DunningDoc createDunningDoc(final I_C_Dunning_Candidate candidate)
 		return doc;
 	}
 
+	@Nullable
+	@VisibleForTesting
+	/*package*/ static String getPOReferenceToUse(final I_C_Dunning_Candidate candidate)
+	{
+		final ISysConfigBL sysConfigBL = Services.get(ISysConfigBL.class);
```

Comment:
sysconfigBL - why not a field?

Reply:
static method

Reply:
ahh ok, but that's an argument to not have it static.

NOTE: i am suggesting to make a method static when I think it's a pure function (i.e. do not depends on state / always uses the parameters only to determine the return value).

In this case, this is not a pure fiction. So it shall not be static.
Also in junit we could instantiate the DefaultDunningProducer and then call it.